tests/functional/test_config_groups.py

```diff
@@ -0,0 +1,152 @@
+
+CLUSTER_BUNDLE_PATH = get_data_dir(__file__, "cluster_simple")
+PROVIDER_BUNDLE_PATH = get_data_dir(__file__, "hostprovider_bundle")
+
+
+@pytest.fixture()
+def cluster(request, sdk_client_fs: ADCMClient) -> Cluster:
+    bundle_path = request.param if hasattr(request, "param") else CLUSTER_BUNDLE_PATH
+    cluster_bundle = sdk_client_fs.upload_from_fs(bundle_path)
+    return cluster_bundle.cluster_create(name=utils.random_string())
+
+
+@pytest.fixture()
+def provider(request, sdk_client_fs: ADCMClient) -> Provider:
+    bundle_path = request.param if hasattr(request, "param") else PROVIDER_BUNDLE_PATH
+    provider_bundle = sdk_client_fs.upload_from_fs(bundle_path)
+    return provider_bundle.provider_create(name=utils.random_string())
+
+
+class TestGroupsIntersection:
+
+    HOST_ERROR_MESSAGE = (
+        "host is not available for this object, or host already is a member of another group of this object"
+    )
+    FIRST_COMPONENT_NAME = "first"
+    SECOND_COMPONENT_NAME = "second"
+    FIRST_GROUP = "test_group"
+    SECOND_GROUP = "test_group_2"
+    FIRST_HOST = "test_host_1"
+    SECOND_HOST = "test_host_2"
+
+    def create_two_hosts(self, provider: Provider):
+        with allure.step("Create host for config groups"):
+            test_host_1 = provider.host_create(fqdn=self.FIRST_HOST)
+        with allure.step("Create host for host candidate check"):
+            test_host_2 = provider.host_create(fqdn=self.SECOND_HOST)
+        return test_host_1, test_host_2
+
+    @allure.step('Check error')
+    def assert_that_host_add_is_unavaliable(self, error):
+        GROUP_CONFIG_HOST_ERROR.equal(error)
+        assert self.HOST_ERROR_MESSAGE in error.value.error['desc']
+
+    @allure.step("Check that the only second host is present in candidates on second provider group")
+    def assert_host_candidate_equal_expected(self, group: List[Host], expected_hosts: int):
+        assert len(group) == expected_hosts, "One host should be available in group"
+        assert group[0].fqdn == self.SECOND_HOST, f"Should be available host '{self.SECOND_HOST}'"
+
+    @pytest.fixture()
+    def cluster_with_components(self, cluster, provider):
+        service = cluster.service_add(name='test_service_1')
+        test_host_1, test_host_2 = self.create_two_hosts(provider)
+        cluster.host_add(test_host_1)
+        cluster.host_add(test_host_2)
+        cluster.hostcomponent_set(
+            (test_host_1, service.component(name=self.FIRST_COMPONENT_NAME)),
+            (test_host_2, service.component(name=self.FIRST_COMPONENT_NAME)),
+            (test_host_2, service.component(name=self.SECOND_COMPONENT_NAME)),
+        )
+        return service, test_host_1, test_host_2
+
+    def test_that_groups_not_allowed_to_intersect_in_cluster(self, sdk_client_fs, cluster, provider):
+        """Test that groups are not allowed to intersect in cluster"""
+
+        test_host_1, test_host_2 = self.create_two_hosts(provider)
```

Comment:
It's not obligatory, but I guess it's better to move such things to "Setup" (fixture) when it's possible.
And I guess it can be used in some fixtures above.
@skhomuti what do you think?

Reply:
yep, i think so too

Reply:
ok, changed, but i see no difference